packages/pyright-scip/src/treeVisitor.ts

```diff
@@ -779,10 +787,14 @@ export class TreeVisitor extends ParseTreeWalker {
                             relationships,
                         })
                     );
+
+                    this.pushNewOccurrence(node, this.getScipSymbol(decl.node), scip.SymbolRole.Definition, parent);
```

Comment:
Not exactly part of your patch, but I think this code would be clearer if the usages of `parent` in this branch would be replaced by `decl.node`. IIUC, that is going to be true, because we're in the `isDefinition = decl.node.is == parent.id` case, but it's a little confusing to read this because there are seemingly 3 different things here (`node`, `decl.node`, `parent`):

```
                    this.pushNewOccurrence(node, this.getScipSymbol(decl.node), scip.SymbolRole.Definition, parent);
```

whereas the code path earlier for enclosing_ranges has:

```
this.pushNewOccurrence(node, existingSymbol, scip.SymbolRole.Definition, decl.node);
```

where `existingSymbol` was derived from `decl.node` just a few lines earlier.